src/main/java/io/confluent/kafkarest/converters/BigDecimalEncoder.java

```diff
@@ -0,0 +1,18 @@
+package io.confluent.kafkarest.converters;
+
+import java.math.BigDecimal;
+import java.nio.charset.Charset;
+
+public class BigDecimalEncoder {
+
+    static public byte[] toByteArray(BigDecimal bigDecimal, int scale) {
+        // will throw an error if rounding is necessary (meaning scale is wrong)
+        return bigDecimal.setScale(scale, BigDecimal.ROUND_UNNECESSARY).unscaledValue().toByteArray();
+    }
+
+
+    static public String toEncodedString(BigDecimal bigDecimal, int scale, int precision) {
+        return new String(toByteArray(bigDecimal,scale), Charset.forName("ISO-8859-1"));
+    }
```

Comment:
That's the only charset that I found that could encode the bytes the way I wanted for a String, so that my tests passed. Unsure why this one works and not others, but hoping for Confluent to let me know